### DISCUSSION TITLE
Error (Xcode): Framework 'opencv_dart' not found

### DISCUSSION THREAD

**Discussion by msmartin4470** [rainyl/opencv_dart#36] [Q&A] [answered]
I am sure this is me not doing something correct, but I am trying to build my iOS application using opencv_dart and I am getting an error.  I tried to follow the readme as best I could and the command to download binaries seemed to work ok. Here is the error I am getting after running flutter build iOS.  Any suggestions help is appreciated thanks.

Failed to build iOS app
Error (Xcode): Framework 'opencv_dart' not found

Error (Xcode): Linker command failed with exit code 1 (use -v to see invocation)

Encountered error while building for device.

**Comment by rainyl** [comment]
Are you building for real devices? If so, please refer to https://github.com/rainyl/opencv_dart/issues/24#issuecomment-2047366823 and README in example

**Reply by msmartin4470** [reply]
Thank you.  I am building for real device.  I have rerun the setup command with the proper architecture, but I am still getting the same output as above.  My flutter project that is using opencv_dart package is actually a plugin that my main flutter application uses.  I have tried including the package and performing setup in both my plugin project and my main project.

**Reply by rainyl** [reply]
@msmartin4470 I didn't test it for a plugin, could you please give me a minimal example to reproduce it?

BTW, is it works in simulator?

**Comment by msmartin4470** [comment]
I created a base app and plugin.  It has really nothing in it at all, but I seem to get the same error.  I tried this with simulator also and had the same error.  Thank you for your help.

https://github.com/msmartin4470/temp_app.git

**Reply by rainyl** [reply]
Confirmed, It's a bug of setup script, will be fixed quickly in v1.0.2+1

**Reply by rainyl** [reply, marked as answer]
v1.0.2+1 has been published, please update to v1.0.2+1 and try again, feel free to leave more message if the problem still exists.

**Reply by msmartin4470** [reply]
I believe this is working now I will do some more testing.  Is your calcHist function limited to 8bit images?  I do a lot of work with 16bit X-ray images and I need to use the calcHist function to get a histogram.  If it is limited to 8bit images is it possible to add 16 bit support?  Thanks

**Reply by rainyl** [reply]
Theoretically there is no limitation, you can use just like [c++ opencv](https://docs.opencv.org/4.x/d6/dc7/group__imgproc__hist.html#ga4b2b5fd75503ff9e6844cc4dcdaed35d), I have tested with the following code:
```dart
final src = cv.Mat.randu(256, 256, cv.MatType.CV_16UC3, low: cv.Scalar.zeros, high: cv.Scalar.all(65535));
final mask = cv.Mat.empty();
final hist = cv.calcHist([src].cvd, [0].i32, mask, [256].i32, [0.0, 256.0].f32);
print(hist.shape);
```
Output:
```bash
[256, 1, 1]
```

 if it doesn't work, please paste the error message and provide sevaral example images so I can try to fix it.

**Reply by msmartin4470** [reply]
Thank you for your help. I was getting an assertion failure, but your example works so I must have had one of my arguments wrong.  Once again I really want to thank you for your help.  Please keep this package up!  I have been using my own custom method channel solution for about a year, but this makes things so much easier for me.

**Reply by rainyl** [reply]
Thanks~ :smile:
